### ISSUE TITLE
rpc.py returns an error on "bdev_nvme_attach_controller" - what does it mean?

### ISSUE THREAD

**Issue by nufosmatic** [spdk/spdk#1516]
$ sudo ../../spdk/scripts/rpc.py bdev_nvme_attach_controller -b NVMe0 -t PCIe -a 0000:8d:00.0/1
request:
{
  "name": "NVMe0",
  "trtype": "PCIe",
  "traddr": "0000:8d:00.0/1",
  "method": "bdev_nvme_attach_controller",
  "req_id": 1
}
Got JSON-RPC error response
response:
{
  "code": -1,
  "message": "Method is allowed in any state in the mask (2), but current state is (1)"
}

## Expected Behavior

The diagnostic ought to be usable. "Method" is not allowed in any "state" in the "mask" - these are not really "concepts" revealed in docs.

## Current Behavior

The diagnostic talks in jargon which is not revealed in docs.

## Possible Solution

I wish I had an answer.

## Steps to Reproduce

Given in summary

## Context (Environment including OS version, SPDK version, etc.)

SPDK 20.04.1 CentOS 7.5

**Comment by jimharris** [comment]
Hi,

I agree the text here is not descriptive enough.  What the text should be saying is something to the effect of "the application is currently in STARTUP phase, but this RPC can only be run in RUNTIME phase".

I suspect you've started your application with --wait-for-rpc, and are trying to run this RPC before the framework_start_init RPC.  If you can provide more details on the application you're running along with command line parameters, we can help get you past this error.

Thanks,

Jim

**Comment by Comphix** [comment]
Raised a patch here https://review.spdk.io/gerrit/c/spdk/spdk/+/3565

And below is the output:

root@storage35:/home/gangcao/storage17spdk# ./scripts/rpc.py bdev_set_options
request:
{
  "bdev_auto_examine": true,
  "method": "bdev_set_options",
  "req_id": 1
}
Got JSON-RPC error response
response:
{
  "code": -1,
  "message": "Method is allowed in the state of Startup, but current state is Runtime."
}

**Comment by nufosmatic** [comment]
I finally found a mundane explanation for this error - it means "this application cannot accept RPC commands." I found this in another issue https://github.com/spdk/spdk/issues/300:

> If you’re wanting to run the bdev fio plugin, or the SPDK bdevperf utility, then RPCs won’t work – you’ll need to create a configuration file with an [Nvme] section to specify the NVMe SSD at 0000:00:02.0.

I was attempting to get an fio_plugin application to do what I wanted. I won't be trying to do this any more.

It would be nice if all of the INI files included were all converted to JSON so that we've have a template from which to create usable JSON files.
